pkg/storage/image_regexp.go

```diff
@@ -0,0 +1,123 @@
+package storage
+
+// TODO(runcom): explain why we need this here!!!
+
+import "regexp"
+
+var (
+	// alphaNumericRegexp defines the alpha numeric atom, typically a
+	// component of names. This only allows lower case characters and digits.
+	alphaNumericRegexp = match(`[a-z0-9]+`)
+
+	// separatorRegexp defines the separators allowed to be embedded in name
+	// components. This allow one period, one or two underscore and multiple
+	// dashes.
+	separatorRegexp = match(`(?:[._]|__|[-]*)`)
+
+	// nameComponentRegexp restricts registry path component names to start
+	// with at least one letter or number, with following parts able to be
```

Comment:
Also have you written tests for these?  It would be nice to have a test fail is someone later edits one of these and messes it up.